README.md

```diff
@@ -17,62 +17,98 @@ As a part of the bridge, a Kafka producer will be responsible for producing mess
+Let's go through each rule in the above example to understand how the MQTT Bridge uses these rules to map MQTT topics to
+Kafka topics:
 
-Let's go through each rule in the above example to understand how the MQTT Bridge uses these rules to map MQTT topics to Kafka topics:
+1. MQTT Topic: `building/(\\w+)/room/(\\d{1,4}).*` -> Kafka Topic: `building_$1` with Kafka Key: `room_$2`
 
-1. MQTT Topic: `building/{building}/room/{room}/#` -> Kafka Topic: `building_{building}_room_{room}`
+   This rule maps MQTT topics of the form `building/{some word}/room/{some number with 4 digits}/#` to the Kafka
+   topic `building_$1`.
+   For example, if the MQTT topic is `building/A/room/1003/floor/2` it will be mapped to the Kafka
+   topic `building_A` with the key `room_1003`.
 
-    This rule maps MQTT topics of the form `building/{building}/room/{room}/#` to the Kafka topic `building_{building}_room_{room}`. 
-    For example, if the MQTT topic is `building/A/room/1/floor/2` it will be mapped to the Kafka topic `building_A_room_1_floor_2`.
+2. MQTT Topic: `sensors/([^/]+)/data` -> Kafka Topic: `sensor_data` with Kafka Key: `null`
 
-2. MQTT Topic: `sensors/+/data` -> Kafka Topic: `sensor_data`
+   This rule maps MQTT topics of the form `sensors/+/data` to the Kafka topic `sensor_data`.
+   For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data`.
+   Because the `KafkaKey` is not defined, the key of the Kafka record will be `null`.
 
-    This rule maps MQTT topics of the form `sensors/+/data` to the Kafka topic `sensor_data`. 
-    For example, if the MQTT topic is `sensors/temperature/data`, it will be mapped to the Kafka topic `sensor_data`.
+3. MQTT Topic: `sensors.*` -> Kafka Topic: `sensor_others` with Kafka Key: `null`
 
-3. MQTT Topic: `sensors/#` -> Kafka Topic: `sensor_others`
+   This rule maps any MQTT topic starting with `sensors/` followed by any number of levels in the hierarchy to the Kafka
+   topic `sensor_others`.
+   For example, if the MQTT topic is `sensors/temperature/living-room`, it will be mapped to the Kafka
+   topic `sensor_others` with the key `sensor`.
+   Because the `KafkaKey` is not defined, the key of the Kafka record will be `null`.
 
-   This rule maps any MQTT topic starting with `sensors/` followed by any number of levels in the hierarchy to the Kafka topic `sensor_others`.
-   For example, if the MQTT topic is `sensors/temperature/living-room`,  it will be mapped to the Kafka topic `sensor_others`.
+4. MQTT Topic: `devices/([^/]+)/data/(\b(all|new|old)\b)` -> Kafka Topic: `device_$1_data` with Kafka Key: `device_$2`
 
+   This rule maps MQTT topics of the form `devices/{some word}/data/{either all, new or old}` to the Kafka
+   topic `device_$1_data`.
+   For example, if the MQTT topic is `devices/thermostat/data/all`, it will be mapped to the Kafka
```

Comment:
```suggestion
   For example, if the MQTT topic is `devices/thermostat/data/all`, it will be mapped to the Kafka topic `device_thermostat_data` with the key `device_all`.
```